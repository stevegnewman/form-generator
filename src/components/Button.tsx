import React from 'react';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: 'primary' | 'secondary';
  fullWidth?: boolean;
  children: React.ReactNode;
}

/**
 * Reusable Button component for basic button styling.
 * Supports primary/secondary styles and full width option.
 */
export const Button: React.FC<ButtonProps> = ({ 
  variant = 'primary', 
  fullWidth = false,
  children, 
  className = '',
  ...props 
}) => {
  const baseStyles = "py-[7px] px-[30px] rounded-3xl transition-all duration-200 font-bold";
  
  const variantStyles = {
    primary: "bg-primary text-white hover:bg-white hover:text-bodyText border-2 border-primary",
    secondary: "bg-white text-bodyText hover:bg-primary hover:text-white border-2 border-primary"
  };
    
  const widthStyles = fullWidth ? "w-full" : "w-auto";

  return (
    <button
      className={`${baseStyles} ${variantStyles[variant]} ${widthStyles} ${className}`}
      {...props}
    >
      {children}
    </button>
  );
}; 