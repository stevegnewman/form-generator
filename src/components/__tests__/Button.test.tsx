import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from '../Button';

describe('Button', () => {
  it('renders with default props', () => {
    render(<Button>Click me</Button>);
    const button = screen.getByRole('button', { name: /click me/i });
    
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('bg-primary');
    expect(button).not.toHaveClass('w-full');
  });

  it('renders with secondary variant', () => {
    render(<Button variant="secondary">Secondary</Button>);
    const button = screen.getByRole('button', { name: /secondary/i });
    
    expect(button).toHaveClass('bg-white');
    expect(button).toHaveClass('text-bodyText');
  });

  it('renders full width when fullWidth prop is true', () => {
    render(<Button fullWidth>Full Width</Button>);
    const button = screen.getByRole('button', { name: /full width/i });
    
    expect(button).toHaveClass('w-full');
  });

  it('handles click events', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click me</Button>);
    
    fireEvent.click(screen.getByRole('button', { name: /click me/i }));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('accepts additional className', () => {
    render(<Button className="test-class">With Class</Button>);
    const button = screen.getByRole('button', { name: /with class/i });
    
    expect(button).toHaveClass('test-class');
  });


  /* Snapshot tests */
  it('matches snapshot for primary button', () => {
    const { container } = render(<Button>Primary Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot for secondary button', () => {
    const { container } = render(
      <Button variant="secondary">Secondary Button</Button>
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot for full width button', () => {
    const { container } = render(
      <Button fullWidth>Full Width Button</Button>
    );
    expect(container).toMatchSnapshot();
  });
}); 