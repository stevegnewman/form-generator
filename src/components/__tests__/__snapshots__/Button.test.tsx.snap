// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot for full width button 1`] = `
<div>
  <button
    class="py-[7px] px-[30px] rounded-3xl transition-all duration-200 font-bold bg-primary text-white hover:bg-white hover:text-bodyText border-2 border-primary w-full "
  >
    Full Width Button
  </button>
</div>
`;

exports[`Button matches snapshot for primary button 1`] = `
<div>
  <button
    class="py-[7px] px-[30px] rounded-3xl transition-all duration-200 font-bold bg-primary text-white hover:bg-white hover:text-bodyText border-2 border-primary w-auto "
  >
    Primary Button
  </button>
</div>
`;

exports[`Button matches snapshot for secondary button 1`] = `
<div>
  <button
    class="py-[7px] px-[30px] rounded-3xl transition-all duration-200 font-bold bg-white text-bodyText hover:bg-primary hover:text-white border-2 border-primary w-auto "
  >
    Secondary Button
  </button>
</div>
`;
